VimeoNetworking/VimeoNetworking/Request.swift

```diff
@@ -164,31 +165,69 @@ public struct Request<ModelType: MappableResponse>
          retryPolicy: RetryPolicy? = nil)
     {
         self.method = method
+        
+        let (path, query) = path.splitLinkString()
         self.path = path
-        self.additionalParameters = parameters ?? [:]
-        self.page = page
-        self.itemsPerPage = itemsPerPage
+        
+        if let query = query,
+            let queryParameters = query.parametersFromQueryString()
+        {
+            var page = page
```

Comment:
These `if let`s look similar enough that I wonder if we could come up with some kind of function that uses generics to do this. Does that seem like it could be a thing? They're all trying to conditionally set a variable and assigning it to a property of the same type.